File: csharp/Behavioral/14-Command/Command/CancelJobCommand.cs
public class CancelJobCommand : ICommand
{
    private readonly JobCard _jobCard;
    private string           _previousStatus = string.Empty;

    public CancelJobCommand(JobCard jobCard) => _jobCard = jobCard;

    public void Execute()
    {
        _previousStatus = _jobCard.Status;
        _jobCard.Status = "Cancelled";
        Console.WriteLine($"  Cancelled {_jobCard.JobId}. Status → Cancelled");
    }

    public void Undo()
    {
        _jobCard.Status = _previousStatus;
        Console.WriteLine($"  Undo cancel: {_jobCard.JobId} Status → {_previousStatus}");
    }
}

File: typescript/Behavioral/14-Command/Command/CancelJobCommand.ts
import { ICommand } from "./ICommand";
import { JobCard } from "../Model/JobCard";

export class CancelJobCommand implements ICommand {
  private previousStatus: string = "";

  constructor(private readonly jobCard: JobCard) {}

  execute(): void {
    this.previousStatus = this.jobCard.status;
    this.jobCard.status = "Cancelled";
    console.log(`  Cancelled ${this.jobCard.jobId}. Status → Cancelled`);
  }

  undo(): void {
    this.jobCard.status = this.previousStatus;
    console.log(
      `  Undo cancel: ${this.jobCard.jobId} Status → ${this.previousStatus}`,
    );
  }
}
